tests/unit/s2n_client_cert_verify_test.c

```diff
@@ -149,6 +149,8 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_config_set_unsafe_for_testing(client_config));
         /* This cipher preference is set to avoid TLS 1.3. */
         EXPECT_SUCCESS(s2n_config_set_cipher_preferences(client_config, "20170210"));
+        /* Enable signature validation to test S2N_CLIENT connection signature for async sign operation */
+        EXPECT_SUCCESS(s2n_config_set_async_pkey_validation_mode(client_config, S2N_ASYNC_PKEY_VALIDATION_STRICT));
```

Comment:
Do we need a negative tests for `S2N_CLIENT` too?

Reply:
I think we can consider negative test case in `s2n_async_pkey_test.c` as a negative test for both `conn->mode` (i.e. S2N_SERVER and S2N_CLIENT), as validation logic doesn't have any dependency upon the s2n_connection.

If you suggest we should still keep it, i can replicate the same in `s2n_client_cert_verify_test.c`

Reply:
I think it would be nice to have it, as we can ensure it's working properly and not just giving a success each time.

Reply:
Done. 
Added a negative test case and tested validity with a static flag in `s2n_client_cert_verify_test`